code/game/objects/structures/crates_lockers/largecrate.dm

```diff
@@ -251,8 +251,81 @@
 	name = "blue barrel"
 	desc = "A blue storage barrel."
 	icon_state = "barrel_blue"
+	var/strap_overlay = "+straps"
 	parts_type = /obj/item/stack/sheet/metal
 	unpacking_sound = 'sound/effects/metalhit.ogg'
+	var/straps = FALSE
+
+/obj/structure/largecrate/random/barrel/true_random
+	name = "barrel"
+	desc = "A barrel."
+	icon_state = "barrel_recolorable"
+	desc_lore = "From the future."
+	var/cap_doodad_state = ""
+	var/center_doodad_state = ""
+	var/color_override = null
+
+
+var/static/list/rbarrel_cap_states = list()
+var/static/list/rbarrel_center_states = list()
+var/global/rbarrel_genned = FALSE
+var/static/list/rbarrel_color_list = list(COLOUR_SILVER,
+	COLOUR_FLOORTILE_GRAY,
+	COLOUR_MAROON,
+	COLOUR_SOFT_RED,
+	COLOUR_LIGHT_GRAYISH_RED,
+	COLOUR_VERY_SOFT_YELLOW,
+	COLOUR_OLIVE,
+	COLOUR_DARK_MODERATE_LIME_GREEN,
+	COLOUR_TEAL,
+	COLOUR_MODERATE_BLUE,
+	COLOUR_PURPLE,
+	COLOUR_STRONG_VIOLET,
+	COLOUR_BEIGE,
+	COLOUR_DARK_MODERATE_ORANGE,
+	COLOUR_BROWN,
+	COLOUR_DARK_BROWN)
```

Comment:
To resolve the linting error, see https://github.com/cmss13-devs/cmss13/pull/4940/files

You'll want to use either `GLOBAL_VAR` or `GLOBAL_LIST` for these and then add the prefix `GLOB.` to any usage of these variables

Reply:
To strictly be a static list only for a barrel, this would need to be instead be defined under `/obj/structure/largecrate/random/barrel` or something. As is since they aren't under any datum they are global.